src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -621,23 +615,13 @@ impl ThinPool {
         } else if meta_extend_failed {
             new_state = PoolExtendState::MetaFailed;
         }
-        if self.extend_state() != new_state {
+        if self.pool_extend_state != new_state {
```

Comment:
Please condense this to ```self.pool_extend_state = new_state``` as you did w/, e.g., ```free_space_state```.

Reply:
Exactly!

Reply:
Resolved.